core/src/test/java/com/datastax/oss/driver/internal/core/context/DefaultDriverContextTest.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.core.context;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.datastax.oss.driver.api.core.config.DefaultDriverOption;
+import com.datastax.oss.driver.api.core.config.DriverExecutionProfile;
+import com.datastax.oss.driver.internal.core.protocol.Lz4Compressor;
+import com.datastax.oss.driver.internal.core.protocol.SnappyCompressor;
+import com.datastax.oss.protocol.internal.Compressor;
+import com.datastax.oss.protocol.internal.NoopCompressor;
+import com.tngtech.java.junit.dataprovider.DataProvider;
+import com.tngtech.java.junit.dataprovider.DataProviderRunner;
+import io.netty.buffer.ByteBuf;
+import java.util.Optional;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(DataProviderRunner.class)
+public class DefaultDriverContextTest {
+
+  private DefaultDriverContext buildMockedContext(Optional<String> compressionOption) {
+
+    DriverExecutionProfile defaultProfile = mock(DriverExecutionProfile.class);
+    when(defaultProfile.isDefined(DefaultDriverOption.PROTOCOL_COMPRESSION))
+        .thenReturn(compressionOption.isPresent());
+    compressionOption.ifPresent(
+        compression -> {
+          when(defaultProfile.getString(DefaultDriverOption.PROTOCOL_COMPRESSION))
+              .thenReturn(compression);
+        });
+    return MockedDriverContextFactory.defaultDriverContext(Optional.of(defaultProfile));
+  }
+
+  private void doCreateCompressorTest(Optional<String> configVal, Class<?> expectedClz) {
+
+    DefaultDriverContext ctx = buildMockedContext(configVal);
+    Compressor<ByteBuf> compressor = ctx.getCompressor();
+    assertThat(compressor).isNotNull();
+    assertThat(compressor).isInstanceOf(expectedClz);
+  }
```

Comment:
I thought about creating a data provider for both params, something along the lines of the following:

```java
  @DataProvider
  public static Object[][] createCompressorDP() {

	  Object[][] rv = new Object[12][2];
	  int idx = 0;
	  for (String name : new String[] { "lz4", "lZ4", "Lz4", "LZ4"}) {
		  rv[idx++] = new Object[] { name, Lz4Compressor.class };
	  }
	  for (String name : new String[] { "snappy", "SNAPPY", "sNaPpY", "SNapPy"}) {
		  rv[idx++] = new Object[] { name, SnappyCompressor.class };
	  }
	  for (String name : new String[] { "none", "NONE", "NoNe", "nONe"}) {
		  rv[idx++] = new Object[] { name, NoopCompressor.class };
	  }
	  return rv;
  }

  @Test
  @UseDataProvider("createCompressorDP")
  public void should_create_compressor(String configVal, Class<?> expectedClz) {

	    DefaultDriverContext ctx = buildMockedContext(Optional.of(configVal));
	    Compressor<ByteBuf> compressor = ctx.getCompressor();
	    assertThat(compressor).isNotNull();
	    assertThat(compressor).isInstanceOf(expectedClz);
	  }
```

This made the test output just a bit harder to read, though, so I settled on the middle ground above.